Jupyter Notebook for mini project 1. This project gives data about the frequency that Berry students eat off campus. 

In [45]:
import csv
import matplotlib.pyplot as plt

In [94]:
def totalEntries():
    entries = 0
    with open("miniProject.csv") as csvFile:
        openCSV = csv.reader(csvFile, delimiter = ',')

        next(openCSV)
        for i in openCSV:
            entries += 1
    return (entries)

print(totalEntries())

35


Total number of entries: 35

In [59]:
def getGenderPercent():
    female = 0
    male = 0
    with open("miniProject.csv") as csvFile:
        openCSV = csv.reader(csvFile, delimiter = ',')

        next(openCSV)
        for row in openCSV:
            gender = row[2]
            if gender == "Female":
                female += 1
            else:
                male += 1
        femalePercent = 100 * (round(female/totalEntries(), 2))
        malePercent = 100 * (round(male/totalEntries(), 2))
    return (f"{femalePercent}% Female", f"{malePercent}% Male")

print(getGenderPercent())

('54.0% Female', '46.0% Male')


Percentages of male vs female: '54.0% Female', '46.0% Male'

In [87]:
def getClassPercent():
    classes = {}

    with open("miniProject.csv") as csvFile:
            openCSV = csv.reader(csvFile, delimiter = ',')

            next(openCSV)
            for row in openCSV:
                Class = row[4].lower().strip()
                if Class not in classes:
                    classes[Class] = 1
                else:
                    classes[Class] += 1
            percentages = {}
            
            for Class, frequency in classes.items():
                percentage = 100 * round((frequency / totalEntries()), 2) 
                percentages[Class] = f"{percentage}%"
    return percentages

print(getClassPercent())

{'senior': '20.0%', 'freshman': '11.0%', 'sophomore': '23.0%', 'junior': '46.0%'}


Percentages of each class surveyed: 'senior': '20.0%', 'freshman': '11.0%', 'sophomore': '23.0%', 'junior': '46.0%'

In [86]:
def getMajorPercent():
    majors = {}

    with open("miniProject.csv") as csvFile:
        openCSV = csv.reader(csvFile, delimiter = ',')

        next(openCSV)
        for row in openCSV:
            major = row[3].lower().strip()
            if major not in majors:
                majors[major] = 1
            else:
                majors[major] += 1
                
        percentages = {}
        for major, frequency in majors.items():
            percentage = 100 * round((frequency / totalEntries()), 2) 
            percentages[major] = f"{percentage}%"
    return percentages
        
print(getMajorPercent())

{'biology': '3.0%', 'finance and accounting': '3.0%', 'biochemistry': '6.0%', 'chemistry': '3.0%', 'exercise science': '14.000000000000002%', 'health sciences': '3.0%', 'creative technologies': '3.0%', 'psychology': '3.0%', 'elementary education': '6.0%', 'computer science and creative technologies': '3.0%', 'marketing': '3.0%', 'finance': '3.0%', 'sports': '3.0%', 'business': '3.0%', 'professional track physics, mathematics': '3.0%', 'political science': '3.0%', 'philosophy, env studies': '3.0%', 'animal science': '3.0%', 'c': '3.0%', 'communication': '3.0%', 'crt': '3.0%', 'sports leadership and strategy': '3.0%', 'green eggs and ham': '3.0%', 'international affairs and spanish': '3.0%', 'accounting & finance': '3.0%', 'economics': '3.0%', 'nursing': '3.0%', 'art': '3.0%', 'art education': '3.0%'}


Percentages of each major that was surveyed: 'biology': '3.0%', 'finance and accounting': '3.0%', 'biochemistry': '6.0%', 'chemistry': '3.0%', 'exercise science': '14.000000000000002%', 'health sciences': '3.0%', 'creative technologies': '3.0%', 'psychology': '3.0%', 'elementary education': '6.0%', 'computer science and creative technologies': '3.0%', 'marketing': '3.0%', 'finance': '3.0%', 'sports': '3.0%', 'business': '3.0%', 'professional track physics, mathematics': '3.0%', 'political science': '3.0%', 'philosophy, env studies': '3.0%', 'animal science': '3.0%', 'c': '3.0%', 'communication': '3.0%', 'crt': '3.0%', 'sports leadership and strategy': '3.0%', 'green eggs and ham': '3.0%', 'international affairs and spanish': '3.0%', 'accounting & finance': '3.0%', 'economics': '3.0%', 'nursing': '3.0%', 'art': '3.0%', 'art education': '3.0%'

In [89]:
def getFrequencyOffCampus():
    freq = {}

    with open("miniProject.csv") as csvFile:
        openCSV = csv.reader(csvFile, delimiter = ',')

        next(openCSV)
        for row in openCSV:
            answer = row[5].lower().strip()
            if answer not in freq:
                freq[answer] = 1
            else:
                freq[answer] += 1

        percentages = {}
        for answer, frequency in freq.items():
            percentage = 100 * round((frequency / totalEntries()), 2) 
            percentages[answer] = f"{percentage}%"
    return percentages

print(getFrequencyOffCampus())

{'2 times a week': '26.0%', '5+ times a week': '14.000000000000002%', '1 time a week': '20.0%', '4 times a week': '11.0%', '3 times a week': '20.0%', '5 times a week': '9.0%'}


Percentages of frequency students eat off campus: '2 times a week': '26.0%', '5+ times a week': '14.%', '1 time a week': '20.0%', '4 times a week': '11.0%', '3 times a week': '20.0%', '5 times a week': '9.0%'

In [92]:
def whenOffCampus():
    freq = {}

    with open("miniProject.csv") as csvFile:
        openCSV = csv.reader(csvFile, delimiter = ',')

        next(openCSV)
        for row in openCSV:
            answer = row[7].lower().strip()
            if answer not in freq:
                freq[answer] = 1
            else:
                freq[answer] += 1

        percentages = {}
        for answer, frequency in freq.items():
            percentage = 100 * round((frequency / totalEntries()), 2) 
            percentages[answer] = f"{percentage}%"
    return percentages

print(whenOffCampus())

{'dinner': '28.999999999999996%', 'breakfast;lunch;dinner': '3.0%', 'breakfast;lunch;dinner;later than 10pm': '6.0%', 'lunch;dinner': '20.0%', 'dinner;later than 10pm': '26.0%', 'breakfast;dinner': '9.0%', 'lunch;dinner;later than 10pm': '6.0%', 'lunch': '3.0%'}


Percentages of when people go off campus: 'dinner': '28.99%', 'breakfast;lunch;dinner': '3.0%', 'breakfast;lunch;dinner;later than 10pm': '6.0%', 'lunch;dinner': '20.0%', 'dinner;later than 10pm': '26.0%', 'breakfast;dinner': '9.0%', 'lunch;dinner;later than 10pm': '6.0%', 'lunch': '3.0%'

In [90]:
def getWhereToEat():
    freq = {}

    with open("miniProject.csv") as csvFile:
        openCSV = csv.reader(csvFile, delimiter = ',')

        next(openCSV)
        for row in openCSV:
            answer = row[9].lower().strip()
            if answer not in freq:
                freq[answer] = 1
            else:
                freq[answer] += 1

        percentages = {}
        for answer, frequency in freq.items():
            percentage = 100 * round((frequency / totalEntries()), 2) 
            percentages[answer] = f"{percentage}%"
    return percentages

print(getWhereToEat())

{'soho': '28.999999999999996%', 'chipotle': '14.000000000000002%', 'panera': '3.0%', "wendy's": '3.0%', 'cookout': '6.0%', 'dos bros': '6.0%', 'taco bell': '9.0%', 'mellow mushroom': '3.0%', 'starbucks & lumina & las palmas': '3.0%', 'anything': '3.0%', 'dos bros‼️🔥': '3.0%', "mcdonald's": '9.0%', 'chick fil a': '3.0%', 'jerusalem grill': '3.0%', 'ru sans': '3.0%', 'pub subs and sushi': '3.0%'}


Percentages of responses of favorite places to go off campus to eat: 'soho': '28.99%', 'chipotle': '14.00%', 'panera': '3.0%', "wendy's": '3.0%', 'cookout': '6.0%', 'dos bros': '6.0%', 'taco bell': '9.0%', 'mellow mushroom': '3.0%', 'starbucks & lumina & las palmas': '3.0%', 'anything': '3.0%', 'dos bros‼️🔥': '3.0%', "mcdonald's": '9.0%', 'chick fil a': '3.0%', 'jerusalem grill': '3.0%', 'ru sans': '3.0%', 'pub subs and sushi': '3.0%'

In [93]:
def getWhyGoOutFreq():
    freq = {}

    with open("miniProject.csv") as csvFile:
        openCSV = csv.reader(csvFile, delimiter = ',')

        next(openCSV)
        for row in openCSV:
            answer = row[6].lower().strip()
            if answer not in freq:
                freq[answer] = 1
            else:
                freq[answer] += 1

        percentages = {}
        for answer, frequency in freq.items():
            percentage = 100 * round((frequency / totalEntries()), 2) 
            percentages[answer] = f"{percentage}%"
    return percentages

print(getWhyGoOutFreq())


{"on campus dining sucks;i don't have any meal swipes": '6.0%', 'on campus dining sucks;i just got money like that': '9.0%', 'on campus dining sucks': '28.999999999999996%', 'celebrations/events': '9.0%', "i don't have any meal swipes": '6.0%', 'i just got money like that': '9.0%', 'on campus dining sucks;celebrations/events': '26.0%', "on campus dining sucks;celebrations/events;i just got money like that;i don't have any meal swipes": '3.0%', 'on campus dining sucks;celebrations/events;i just got money like that': '3.0%', '': '3.0%'}


Percentages of responses as to why students go out to eat: "on campus dining sucks;i don't have any meal swipes": '6.0%', 'on campus dining sucks;i just got money like that': '9.0%', 'on campus dining sucks': '28.99%', 'celebrations/events': '9.0%', "i don't have any meal swipes": '6.0%', 'i just got money like that': '9.0%', 'on campus dining sucks;celebrations/events': '26.0%', "on campus dining sucks;celebrations/events;i just got money like that;i don't have any meal swipes": '3.0%', 'on campus dining sucks;celebrations/events;i just got money like that': '3.0%', '': '3.0%'

In [95]:
def getWhyGoOutThere():
    freq = {}

    with open("miniProject.csv") as csvFile:
        openCSV = csv.reader(csvFile, delimiter = ',')

        next(openCSV)
        for row in openCSV:
            answer = row[10].lower().strip()
            if answer not in freq:
                freq[answer] = 1
            else:
                freq[answer] += 1

        percentages = {}
        for answer, frequency in freq.items():
            percentage = 100 * round((frequency / totalEntries()), 2) 
            percentages[answer] = f"{percentage}%"
    return percentages

print(getWhyGoOutThere())

{'i just love it': '23.0%', 'good food;close by': '3.0%', 'good food;close by;i just love it': '9.0%', 'good food;cheap;i just love it': '3.0%', 'i just love it;it’s so bussy in my tummy': '3.0%', 'good food;close by;cheap;i just love it': '14.000000000000002%', 'good food;cheap': '11.0%', 'good food;i just love it': '9.0%', 'close by;cheap;i just love it': '6.0%', 'good food': '11.0%', 'cheap;i just love it': '3.0%', 'cheap': '3.0%', 'good food;close by;cheap;i just love it;dhall is not appealing to my tastebuds or stomach. i have eaten 3 actual full meals there this semester. each time i feel horrible afterwards… eating is not an enjoyable experience in dhall.': '3.0%'}


Percentages of responses as to why they like the places they go out to eat over on campus dining: 'i just love it': '23.0%', 'good food;close by': '3.0%', 'good food;close by;i just love it': '9.0%', 'good food;cheap;i just love it': '3.0%', 'i just love it;it’s so bussy in my tummy': '3.0%', 'good food;close by;cheap;i just love it': '14.00%', 'good food;cheap': '11.0%', 'good food;i just love it': '9.0%', 'close by;cheap;i just love it': '6.0%', 'good food': '11.0%', 'cheap;i just love it': '3.0%', 'cheap': '3.0%', 'good food;close by;cheap;i just love it;dhall is not appealing to my tastebuds or stomach. i have eaten 3 actual full meals there this semester. each time i feel horrible afterwards… eating is not an enjoyable experience in dhall.': '3.0%'

In [97]:
def getWhyGoOutThen():
    freq = {}

    with open("miniProject.csv") as csvFile:
        openCSV = csv.reader(csvFile, delimiter = ',')

        next(openCSV)
        for row in openCSV:
            answer = row[8].lower().strip()
            if answer not in freq:
                freq[answer] = 1
            else:
                freq[answer] += 1

        percentages = {}
        for answer, frequency in freq.items():
            percentage = 100 * round((frequency / totalEntries()), 2) 
            percentages[answer] = f"{percentage}%"
    return percentages

print(getWhyGoOutThen())

{"on campus food at this time isn't what i'm looking for": '37.0%', 'everything on campus is closed': '3.0%', "i'm already off campus": '6.0%', "everything on campus is closed;on campus food at this time isn't what i'm looking for": '28.999999999999996%', "everything on campus is closed;i'm already off campus": '6.0%', "everything on campus is closed;in between when they are serving real meals;on campus food at this time isn't what i'm looking for": '3.0%', "in between when they are serving real meals;on campus food at this time isn't what i'm looking for": '6.0%', "everything on campus is closed;i'm already off campus;on campus food at this time isn't what i'm looking for": '3.0%', "in between when they are serving real meals;i'm already off campus;on campus food at this time isn't what i'm looking for": '3.0%', "i'm already off campus;on campus food at this time isn't what i'm looking for": '3.0%', "in between when they are serving real meals;i'm already off campus": '3.0%'}


Percentages of responses as to why they go out when they do: "on campus food at this time isn't what i'm looking for": '37.0%', 'everything on campus is closed': '3.0%', "i'm already off campus": '6.0%', "everything on campus is closed;on campus food at this time isn't what i'm looking for": '28.99%', "everything on campus is closed;i'm already off campus": '6.0%', "everything on campus is closed;in between when they are serving real meals;on campus food at this time isn't what i'm looking for": '3.0%', "in between when they are serving real meals;on campus food at this time isn't what i'm looking for": '6.0%', "everything on campus is closed;i'm already off campus;on campus food at this time isn't what i'm looking for": '3.0%', "in between when they are serving real meals;i'm already off campus;on campus food at this time isn't what i'm looking for": '3.0%', "i'm already off campus;on campus food at this time isn't what i'm looking for": '3.0%', "in between when they are serving real meals;i'm already off campus": '3.0%'